### PR TITLE
Compiler: alias analysis speedup

### PR DESCRIPTION
**Pull request by frej**
This is a set of patches which speeds up the alias analysis pass by reducing the number of iterations required to find a fix point and reduces the amount of work in each iteration.

### CODE REVIEW

**Comment by RobinMorisset** [comment]
This looks interesting! Have you done any benchmarking confirming that it is a performance win?

File: lib/compiler/src/beam_ssa_alias.erl [hunk 1]
```diff
@@ -325,11 +330,22 @@ aa_fun(F, #opt_st{ssa=Linear0,anno=_Anno,args=Args}=St,
...
+    ?DP("SS:~n~s~n~n", [SS]),
     AAS = aa_merge_call_args_status(SS, AAS1),
 
     AliasMap = AliasMap0#{ F => SS },
-    {St#opt_st{ssa=Linear1}, AAS#aas{alias_map=AliasMap}}.
+    PrevSS = maps:get(F, AliasMap0, #{}),
+    Repeats = case PrevSS =/= SS of
+                  true ->
+                      %% Alias status has changed, so schedule both
+                      %% our callers and callees for renews analysis.
```
Comment (bjorng) [inline lib/compiler/src/beam_ssa_alias.erl:341]:
```suggestion
                      %% our callers and callees for renewed analysis.
```

**Review by bjorng — APPROVED** [review]
Looks good, except for a typo in a comment.

File: lib/compiler/src/beam_ssa_alias.erl (hunk 1 above)
Comment (frej) [reply lib/compiler/src/beam_ssa_alias.erl]: Fixed and force-pushed.

**Comment by bjorng** [comment]
Thanks! Added to our daily builds.

**Comment by frej** [comment]
> This looks interesting! Have you done any benchmarking confirming that it is a performance win?

For the set of modules compiled by the `diffable`-script these patches reduces the time spent in the alias pass with around 20%.

**Comment by bjorng** [comment]
Thanks for your pull request.

**Pull request merged**